### PR TITLE
refactoring: edit tasks 15 and 18 with tests

### PR DESCRIPTION
**Pull request by diamandv**

### CODE REVIEW

File: src/main/java/SomeTwentyMethods.java [hunk 1]
```diff
@@ -271,10 +271,12 @@ public int[] getIntArrayOfMultipliedValues(int[] integers1, int[] integers2) {
         }
         int[] result = new int[integers1.length];
```
Comment (BogdanParashchak) [inline src/main/java/SomeTwentyMethods.java:272]: can it be done on-fly
try not create new array but insert new values in existing array

**Review by BogdanParashchak — CHANGES_REQUESTED** [review]

File: src/main/java/SomeTwentyMethods.java [hunk 2]
```diff
@@ -271,10 +271,12 @@ public int[] getIntArrayOfMultipliedValues(int[] integers1, int[] integers2) {
...
         for (int i = 0; i < integers1.length; i++) {
-            try {
-                result[i] = Math.multiplyExact(integers1[i], integers2[i]);
-            } catch (ArithmeticException e) {
-                throw new ArithmeticException("cannot multiply: " + integers1[i] + " and " + integers2[i] + " at index " + i);
+            long longValue1 = integers1[i];
+            long longValue2 = integers2[i];
+            if (longValue1 * longValue2 > Integer.MAX_VALUE || longValue1 * longValue2 < Integer.MIN_VALUE) {
```
[the remark this answers is not in the document]
Comment (diamandv) [reply src/main/java/SomeTwentyMethods.java:276]: done

File: src/main/java/SomeTwentyMethods.java (hunk 1 above)
Comment (diamandv) [reply src/main/java/SomeTwentyMethods.java:272]: done

File: src/main/java/SomeTwentyMethods.java [hunk 3]
```diff
@@ -316,16 +318,25 @@ public int[] getArrayOfIntegersWithValuesInReverseOrder(int[] intArray) {
...
+            newArrInt[i] = randomInt;
+        }
+        return newArrInt;
     }
 
     // Task 19 - принимает 2 массива чаров, проверяет есть ли в 1 массиве,
-// такая же последовательность символов которую представляет собой второй массив. Возвращает булеан
+    // такая же последовательность символов которую представляет собой второй массив. Возвращает булеан
     public boolean compareTwoCharArrays(char[] charArray1, char[] charArray2) {
         if (checkIfTwoCharArraysAreNull(charArray1, charArray2)) {
```
[the remark this answers is not in the document]
Comment (diamandv) [reply src/main/java/SomeTwentyMethods.java:341]: done

File: src/main/java/SomeTwentyMethods.java [hunk 4]
```diff
@@ -316,16 +318,25 @@ public int[] getArrayOfIntegersWithValuesInReverseOrder(int[] intArray) {
...
+            int randomInt = random.nextInt(upperBorder - lowerBorder + 1) + lowerBorder;
+            newArrInt[i] = randomInt;
+        }
+        return newArrInt;
     }
 
     // Task 19 - принимает 2 массива чаров, проверяет есть ли в 1 массиве,
-// такая же последовательность символов которую представляет собой второй массив. Возвращает булеан
+    // такая же последовательность символов которую представляет собой второй массив. Возвращает булеан
     public boolean compareTwoCharArrays(char[] charArray1, char[] charArray2) {
```
[the remark this answers is not in the document]
Comment (diamandv) [reply src/main/java/SomeTwentyMethods.java]: done

File: src/test/java/SomeTwentyMethodsTest.java [hunk 5]
```diff
@@ -1132,4 +1140,70 @@ void givenNullIntArray_whenGetArrayOfIntegersWithValuesInReverseOrder_thenReturn
...
+    void givenNegativeSizeOfIntegersWithLowerAndUpperBorder_whenGetRandomArrayWithBounds_thenReturnEmptyArray() {
+        //given
+        int sizeOfIntArray = -15;
+        int lowerBorder = 10;
+        int upperBorder = 7;
+        //when
+        int[] actualIntegers = someTwentyMethods.getRandomArrayWithBounds(sizeOfIntArray, lowerBorder, upperBorder);
+        //then
+        assertEquals(0, actualIntegers.length);
+    }
```
[the remark this answers is not in the document]
Comment (diamandv) [reply src/test/java/SomeTwentyMethodsTest.java]: done

File: src/test/java/SomeTwentyMethodsTest.java [hunk 6]
```diff
@@ -1132,4 +1140,237 @@ void givenNullIntArray_whenGetArrayOfIntegersWithValuesInReverseOrder_thenReturn
...
+        int lowerBorder = 0;
+        int upperBorder = 7;
+        //when
+        int[] actualIntegers = someTwentyMethods.getRandomArrayWithBounds(sizeOfIntArray, lowerBorder, upperBorder);
+        //then
+        for (int integer : actualIntegers
+        ) {
+            assertTrue(integer >= lowerBorder);
+            assertTrue(integer <= upperBorder);
+        }
```
Comment (BogdanParashchak) [inline src/test/java/SomeTwentyMethodsTest.java:1224]: formatting:
        for (int integer : actualIntegers) {
            assertTrue(integer >= lowerBorder);
            assertTrue(integer <= upperBorder);
        }

**Review by BogdanParashchak — CHANGES_REQUESTED** [review]

File: src/test/java/SomeTwentyMethodsTest.java [hunk 7]
```diff
@@ -1132,4 +1140,236 @@ void givenNullIntArray_whenGetArrayOfIntegersWithValuesInReverseOrder_thenReturn
...
+        int sizeOfIntArray = 7000;
+        int lowerBorder = 0;
+        int upperBorder = 7;
+        //when
+        int[] actualIntegers = someTwentyMethods.getRandomArrayWithBounds(sizeOfIntArray, lowerBorder, upperBorder);
+        //then
+        for (int integer : actualIntegers) {
+            assertTrue(integer >= lowerBorder);
+            assertTrue(integer <= upperBorder);
+        }
```
Comment (diamandv) [inline src/test/java/SomeTwentyMethodsTest.java:1223]: formated

**Review by BogdanParashchak — APPROVED** [review]

**Pull request merged**